### PR TITLE
filter out own store's name ID

### LINKED ISSUE #699: Can make customer invoice to own store
Build Number: 2.1.0

Description: title

Reproducible: yes

Reproduction Steps:
Make the mobile store's name visible to itself on server
Make invoice to self
Show's up as supplier invoice
Infinite supply?

Comments:

### PR DESCRIPTION
**Pull request by Chris-Petty**
Fixes #699

### CODE REVIEW

File: src/database/UIDatabase.js [hunk 1]
```diff
@@ -33,17 +46,36 @@ export class UIDatabase {
     }
   }
 
-  addListener(...args) { return this.database.addListener(...args); }
```
Comment (Chris-Petty) [inline src/database/UIDatabase.js]: Below is just prettier having a party.

File: src/database/UIDatabase.js [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
       case 'CustomerInvoice':
         // Only show invoices generated from requisitions once finalised
-        return results.filtered('type == "customer_invoice"'
-                                + ' AND (linkedRequisition == null OR status == "finalised")');
+        return results.filtered(
+          'type == "customer_invoice" AND (linkedRequisition == null OR status == "finalised")',
+        );
       case 'SupplierInvoice':
-        return results.filtered('type == "supplier_invoice"'
-                                + ' AND otherParty.type != "inventory_adjustment"');
```
Comment (Chris-Petty) [inline src/database/UIDatabase.js]: Prettier

File: src/database/UIDatabase.js [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+      .objects('Setting')
+      .filtered('key == $0', THIS_STORE_NAME_ID)[0];
+    // Check ownStoreIdSetting exists, won't if not initialised
+    const thisStoreId = thisStoreIdSetting && thisStoreIdSetting.value;
+
     switch (type) {
       case 'CustomerInvoice':
         // Only show invoices generated from requisitions once finalised
-        return results.filtered('type == "customer_invoice"'
-                                + ' AND (linkedRequisition == null OR status == "finalised")');
```
Comment (Chris-Petty) [inline src/database/UIDatabase.js]: Prettier

Comment (Chris-Petty) [reply src/database/UIDatabase.js]: Actually I removed the concatenation, same with supplier invoice.

File: src/database/UIDatabase.js [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
     this.database = database;
   }
 
   objects(type) {
     const results = this.database.objects(translateToCoreDatabaseType(type));
+    const thisStoreIdSetting = this.database
+      .objects('Setting')
+      .filtered('key == $0', THIS_STORE_NAME_ID)[0];
+    // Check ownStoreIdSetting exists, won't if not initialised
+    const thisStoreId = thisStoreIdSetting && thisStoreIdSetting.value;
```
Comment (andreievg) [inline src/database/UIDatabase.js]: can this be named to reflect it's storeNameID not storeID please

**Pull request merged**